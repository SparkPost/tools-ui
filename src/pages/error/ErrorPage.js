import React from 'react';
import { ERROR_MESSAGES } from './constants';

import './ErrorPage.scss';

const ErrorPage = (props) => {
  const { errorCode } = props.params;
  const code = ERROR_MESSAGES[errorCode] ? errorCode : '404';

  return (
    <div className='flex center-xs'>
      <div className='col-xs-12 errorPage'>
        <h5 className='errorPage__desc'>{ERROR_MESSAGES[code]}</h5>
        <h1 className='errorPage__code'>{code}</h1>
      </div>
    </div>
  );
};

export default ErrorPage;
