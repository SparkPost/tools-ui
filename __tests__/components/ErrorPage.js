import React from 'react';
import renderer from 'react-test-renderer';
import ErrorPage from 'pages/error/ErrorPage';

describe('Error page', () => {

  test('should render 404', () => {
    // Probably want to mock react-router somehow
    // This is equivalent to visiting the url - /something
    const mockParams = { 'errorCode': 'something' };
    expect(renderer.create(<ErrorPage params={mockParams}></ErrorPage>)).toMatchSnapshot();
  });

});
