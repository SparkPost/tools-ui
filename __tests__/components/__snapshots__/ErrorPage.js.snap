exports[`Error page should render 404 1`] = `
<div
  className="flex center-xs">
  <div
    className="col-xs-12 errorPage">
    <h5
      className="errorPage__desc">
      The page you are looking for can\'t be found.
    </h5>
    <h1
      className="errorPage__code">
      404
    </h1>
  </div>
</div>
`;
